spring-integration-feed/src/main/java/org/springframework/integration/feed/inbound/FeedEntryMessageSource.java

```diff
@@ -178,6 +179,24 @@ private SyndEntry getNextEntry() {
 	private void populateEntryList() {
 		SyndFeed syndFeed = this.getFeed();
 		if (syndFeed != null) {
+			
+			/* 
+			 * 
+			 * QUESTION:
+			 * shouldn't there be a check here to see if the feed itself has been 
+			 * published/updated since last check? rfc4287 states that "atom:feed 
+			 * elements MUST contain exactly one atom:updated element." SyndFeed
+			 * doesnt seem to have a #getUpdatedDate method but since published
+			 * date is optional in spec, maybe the #getPublishDate takes care of 
+			 * this? My guess that why this is not happening is due to some oddity
+			 * with different implementations of the ATOM specification?
+			 * 
+			 * if (syndFeed.getPublistDate().getTime() > this.lastTime) {
+			 *   ...
+			 * }
```

Comment:
You can take it out or leave as a TODO. It was really left to be answered by the subject matter expert. The ATOM spec says a date should be there, but the RSS spec is a bit different. Just thought some cycles should be saved if they could be.